CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+# TutorialKit Contributing Guide
+
+Hi! We are really excited that you are interested in contributing to TutorialKit. Before submitting your contribution, please make sure to take a moment and read through the following guide:
+
+## Repo Setup
+
+The TutorialKit repo is a monorepo using pnpm workspaces. The package manager used to install and link dependencies must be [pnpm](https://pnpm.io/).
+
+To develop and test packages:
+
+1. Run `pnpm install` in project's root folder
+
+2. Run `pnpm run build` to build sources
+
+3. Run `pnpm run dev` to build sources in watch mode
+
+4. Run `pnpm run test` to run core tests
+
+## Testing TutorialKit against external packages
+
+You may wish to test your locally-modified copy of TutorialKit against another package that is using it. For pnpm, after building TutorialKit, you can use [`pnpm.overrides`](https://pnpm.io/package_json#pnpmoverrides). Please note that `pnpm.overrides` must be specified in the root `package.json` and you must first list the package as a dependency in the root `package.json`:
+
+```json
+{
+  "pnpm": {
+    "overrides": {
+      "@tutorialkit/astro": "file:../tutorialkit/packages/astro",
+      "@tutorialkit/components-react": "file:../tutorialkit/packages/components/react",
+      "@tutorialkit/runtime": "file:../tutorialkit/packages/runtime",
+      "@tutorialkit/theme": "file:../tutorialkit/packages/theme",
+      "@tutorialkit/types": "file:../tutorialkit/packages/types"
+    }
+  }
+}
```

Comment:
This works somewhat ok but is quite far from ideal. Due to `file:` usage, the project must re-run `pnpm install` after TutorialKit packages have updated.

Using `link:` make changes automatically visible, but I was unable to solve issue where TutorialKit's dependencies brought multiple copies of React into the run time and caused hooks to fail. With webpack this could be solved using `resolve.alias` but it just doesn't work with Vite.